### ISSUE TITLE
Error pushing backend updates after modifying @connection field in the schema

### ISSUE THREAD

**Issue by ray415** [aws-amplify/amplify-cli#8368]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### Amplify CLI Version

6.1.1

### What operating system are you using?

Mac

### Amplify Categories

auth, storage, function, api, hosting

### Amplify Commands

push

### Describe the bug

There's a blocking issue when replacing `@connection` directive with `@function` directive and pushing updates to the cloud.

Initially, I had the API schema like this:

```
type ProductGroup
    @model
    @key(
        name: "..."
        fields: ["..."]
        queryField: "..."
    )
    @auth(
        rules: [
            { allow: groups, groups: ["...", "..."] }
            { allow: groups, groupsField: "...", operations: [read] }
        ]
    ) {
    updatedBy: ID
    updatedAt: AWSDateTime
    id: ID
...
    product: Product @connection
...
}
```

So I needed to update the field "product" not to be connected anymore to the 'Product' model but to call the Lambda function that handles the data and the new schema looked like this:
```
type ProductGroup
    @model
    @key(
        name: "..."
        fields: ["..."]
        queryField: "..."
    )
    @auth(
        rules: [
            { allow: groups, groups: ["...", "..."] }
            { allow: groups, groupsField: "...", operations: [read] }
        ]
    ) {
    updatedBy: ID
    updatedAt: AWSDateTime
    id: ID
...
    product: SomeProduct @function(name: "SomeResolver-${env}") 
...
}
```

After that `amplify push` worked fine but when I decided to switch back to the `@connection` or remove the product field entirely the amplify push will fail throwing an error:

`....-20190502193839-....-AZBRWSS9N6CU-FunctionDirectiveStack-60LHSAFFSWX8 AWS::CloudFormation::Stack Wed Oct 06 2021 11:57:14 GMT+0300 Export ....:GetAtt:....ResolversLambdaDataSource:FunctionId cannot be deleted as it is in use by ....-20190502193839-....Client-AZBRWSS9N6CU-ConnectionStack-IMWH6COIO4PX`

So basically it looks like this Lambda function is being referred to in ConnectionStack (probably because I used the same field name but removed the `@connection` directive) and so the FunctionDirectiveStack update task is throwing errors.

The same kind of problem is described also [here](https://stackoverflow.com/questions/69355367/aws-amplify-cannot-revert-from-function-directive-to-connection-directive) and [here](https://forums.aws.amazon.com/message.jspa?messageID=997006#997006).

Is it possible to somehow manually remove the reference to the Lambda function inside of ConnectionStack so that I could successfully remove the 'product' field and then add it back later on?

### Expected behavior

If the @connection directive (not the field itself) is removed from the schema then it should clean up the ConnectionStack properly.

### Reproduction steps

1. Add `@connection` directive to some schema field and do `amplify push`
2. Replace the `@connection` directive with `@function` directive and do `amplify push`
3. Try now to remove this field or switch back to `@connection` and do `amplify push`

### Additional information

Unfortunately, that's another point to the category why not use Amplify in production because this issue probably prevents rolling back to some previous commits when updating the API will fail.

**Comment by yuth** [comment]
@ray415 to unblock could you do iterative deployments by 
1. Removing the `@connection` directive first from the `product` field 
2. Do an `amplify push`
3. Add the `@function` directive to product field
4. Do another push.


It would be helpful to have the complete project including the stack mapping if any to debug this issue. Could you share your project and by sending email to amplify-cli[@]amazon.com. Please redact any sensitive information from the project before sending it over.

**Comment by ray415** [comment]
@yuth any ideas?

**Comment by josefaidt** [comment]
Hey @ray415 :wave: apologies for the delay! Are you still experiencing this issue with the [latest version of the CLI](https://github.com/aws-amplify/amplify-cli/releases/tag/v7.5.5)?

**Comment by cjihrig** [comment]
Closing due to lack of follow up. If you are still having problems please open a new issue.
